### ISSUE TITLE
LNbank can not pay invoice without amount

### ISSUE THREAD

**Issue by iBobik** [btcpayserver/btcpayserver-plugins#23]
I tried to paste invoice from Phoenix without specified amount expecting it will send all sats:

<img width="649" alt="Snímek obrazovky 2022-05-04 v 20 17 54" src="https://user-images.githubusercontent.com/614232/166800892-59435905-71ea-453d-ab49-fe3610dfa1de.png">

Logs:
```
fail: BTCPayServer.Plugins.LNbank.Pages.Wallets.SendModel: Payment failed
HTTP Response: 

{"code":2,"message":"amount must be specified when paying a zero amount invoice","details":[]}

BTCPayServer.Lightning.LND.SwaggerException: The HTTP status code of the response was not expected (500).
```

**Comment by dennisreimann** [comment]
Hmm, I'm not sure this is wanted or what the user intends. Plainly sending all sats in case the invoice does not contain an amount sounds weird to me.

**Comment by iBobik** [comment]
It is hard to send all sats because it is hard to predict fees, this is why
Bitcoin wallets always had Send all button. In LN empty amount in only
practical way how to allow sender send all.

Btw, all mobile LN wallets can work with this kind of invoices, most has
Send all button.


Honza Pobořil


4. 5. 2022 21:57:44, d11n ***@***.***> napsal:

**Comment by dennisreimann** [comment]
Ok, I get that - I think it would need a warning or so, to distinguish it from the regular use case. Thanks for elaborating, I'll try to allow for that.

**Comment by dennisreimann** [comment]
We'll need a new Lightning lib (and hence BTCPay Server update) for this, but this will be fixed once btcpayserver/BTCPayServer.Lightning#71 is merged.

**Comment by dennisreimann** [comment]
Fixed with [v1.2.2](https://github.com/dennisreimann/btcpayserver/releases/tag/BTCPayServer.LNbank%2Fv1.2.2).
